internal/pkg/xds/adsc/adsc.go

```diff
@@ -80,6 +80,15 @@ func (a *ADSC) Run() error {
 	return nil
 }
 
+func (a *ADSC) Restart() {
+	log.Infof("reconnecting to ADS server %s", a.cfg.DiscoveryAddr)
+	err := a.Run()
+	if err != nil {
+		log.Errorf("failed to Restart to ADS server %s: %v", a.cfg.DiscoveryAddr, err)
+		time.AfterFunc(reconnectDelay, a.Restart)
+	}
```

Comment:
General idea related to comment above:
```suggestion
func (a *ADSC) Restart(retryCount int) error {
	log.Infof("reconnecting to ADS server %s with %d attempt(s) remaining", a.cfg.DiscoveryAddr, retryCount)
	err := a.Run()
	if err != nil {
		if retryCount == 1 {
            return fmt.Errorf("failed to Restart ADS client %s: %w", a.cfg.DiscoveryAddr, err)
        }
	    restartFunc := func() { a.Restart(retryCount-1) }
		time.AfterFunc(reconnectDelay, restartFunc)
	}
```

Reply:
`Restart` is also used when a connection is interrupted in `handleRecv`, and we should always restart the client in such a case.